app/code/Magento/Inventory/Model/ResourceModel/PredefinedId.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model\ResourceModel;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Model\AbstractModel;
+
+/**
+ * Provide possibility of save entity with predefined id
+ * Use trait instead of extending for better reusability, also we don'nt want to hardcode hierarchy of inheritance
```

Comment:
The choice to use trait instead of inheritance was made to prevent the introduction of new layer super type on the module basis as well as better code reusability, as potentially current trait not coupled to Inventory module and other modules could re-use this approach.